Tms/merchant revenue #13

tmikeschu opened the pull request:
Worked with @Dpalazzari - ready for review

- Adds endpoint for merchant revenue
- Adds model instance methods for those endpoints

* Room to potentially refactor the Merchant#revenue... methods with a scope/lambda..

Needs in-depth review. All tests passing.

Closes #13 
Closes #20 
Closes #14 
(14 and 20 are duplicates)

@sallymacnicholas

tmikeschu committed:
Add Merchant#successful_transactions

tmikeschu committed:
Add route for merchant revenue

tmikeschu committed:
Add revenue controller

tmikeschu committed:
Add Merchant#revenue methods

tmikeschu committed:
Add passing tests for merchant revenue endpoints

tmikeschu committed:
Add model tests for revenue instance methods

tmikeschu committed:
Merge branch 'master' of github.com:tmikeschu/rails_engine into tms/merchant-revenue-#13

Dpalazzari commented:
@tmikeschu Is this just the method we worked on with Toni, right?

tmikeschu commented:
@Dpalazzari yes, plus the other spec expectation for passing a date query parameter

tmikeschu commented:
Thanks @ToniRib !

ToniRib commented:
Looks like you guys put this together pretty well!

tmikeschu committed:
Resolve merge conflicts

tmikeschu committed:
Merge branch 'master' of github.com:tmikeschu/rails_engine into tms/merchant-revenue-#13

tmikeschu committed:
Restore deleted revenue path

Add forward slashes to routes

Dpalazzari commented:
@tmikeschu  Does this revenue method convert cents into dollars? Does it pass Spec harness?

tmikeschu commented:
@Dpalazzari good catch. I'll add the dollars string conversion and check the spec harness.

Dpalazzari commented on spec/requests/api/v1/merchants/revenue_intelligence_spec.rb:
```diff
@@ -0,0 +1,40 @@
+      get "/api/v1/merchants/#{merchant.id}/revenue?date=#{new_date}"
+
+      expect(response).to be_success
+      expect(response.body).to eq('400')
```
@tmikeschu Should we stick to Toni's convention is make sure the (response.body).to include('400')?

tmikeschu replied:
@Dpalazzari already on this!

tmikeschu committed:
Add json matcher for rspec

tmikeschu committed:
Add formatting for revenue methods

tmikeschu committed:
Adjust test expectations to match formatting

tmikeschu committed:
Refactor location of hash formatting to controller

tmikeschu commented:
@Dpalazzari let's wait until we have serializers going so we can format the revenue values

tmikeschu committed:
Merge branch 'master' into tms/merchant-revenue-#13

tmikeschu committed:
Resolve merge conflicts

tmikeschu committed:
Merge branch 'master' into tms/merchant-revenue-#13

tmikeschu committed:
Resolve merge conflicts

tmikeschu committed:
Merge branch 'master' of github.com:tmikeschu/rails_engine into tms/merchant-revenue-#13

tmikeschu committed:
Adjust invoice item unit price formatting

Changes to match spec harness expectations

tmikeschu committed:
Refactor revenue formatting to revenue serializer

tmikeschu commented:
Worked with @Dpalazzari to polish off this PR. Ready for merge.

- Added a custom serializer for formatting revenue.
- Adjusted formatting for invoice item price

Needs in-depth review, all tests passing AND endpoints and relationships in the spec harness are all passing, along with the two analytics methods.
